src/olympia/accounts/views.py

```diff
@@ -429,6 +434,14 @@ def profile(self, request, *args, **kwargs):
         self.kwargs['pk'] = unicode(self.request.user.pk)
         return self.retrieve(request, *args, **kwargs)
 
+    def perform_destroy(self, instance):
+        if instance.is_developer:
+            raise serializers.ValidationError(ugettext(
+                u'Developers of add-ons or themes cannot delete their '
+                u'account.  You must delete, or transfer to other users, all '
+                u'add-ons and themes linked to this account first.'))
```

Comment:
The whole sentence sounds weird isn't "you must delete or transfer all add-ons and themes linked to this account to other users first." better I wonder?